src/main/java/io/cdap/plugin/ariba/source/AribaServices.java

```diff
@@ -110,8 +111,26 @@ public class AribaServices {
   private boolean isMinuteLimitExhausted;
   private boolean isSecondsLimitExhausted;
 
-  public AribaServices(AribaConnectorConfig pluginConfig) {
+  private final Integer initialRetryDuration;
+  private final Integer maxRetryDuration;
+  private final Integer maxRetryCount;
+  private final Integer retryMultiplier;
+
+  /**
+   * Determines whether to trigger exponential retry immediately without waiting for the Ariba response retry delay.
+   * If set to true, exponential retry will occur without considering the scheduled delay.
+   */
+  private final boolean immediateRetryRequired;
```

Comment:
In which case is this required?

Reply:
It's set to true in `AribaBatchSource.java`

We have 2 kind of waits, one that waits when api sends back a wait for header and the other is where we are waiting during an exponential back-off policy of the retry.


We try to skip the wait suggested by api, cause sometime this may wait for minutes, causing a timeout on UI.
We still have the exponential back-off wait, if even that fails we show a correct error to user.

now failure may happen for some other reason, we saw if we skip the api waits but not the back-off waits we get the best results.

Reply:
When it is a synchronous operation initiated by user, it is better to give a result back as soon as possible.

I don't have a strong opinion here and it is more of a UX question whether we want to retry during design time operation or return the response if there is a `rateLimitExceeded`, do we have an approval from product (@deepinder)?

Reply:
@itsankit-google in our discussion with deepinder he said it's okay to have this at design time, we have similar logic in other SAP plugin !